atlasdb-api/src/main/java/com/palantir/atlasdb/config/LeaderConfig.java

```diff
@@ -47,15 +49,13 @@ public File acceptorLogDir() {
     public abstract String localServer();
 
     @Size(min = 1)
-    public abstract Set<String> leaders();
+    public abstract List<String> leaders();
 
     @JsonProperty("lockCreator")
     @Value.Default
     public String lockCreator() {
-        if (leaders().isEmpty()) {
-            throw new IllegalStateException("The leaders block cannot be empty");
-        }
-        return leaders().iterator().next();
+        Preconditions.checkState(!leaders().isEmpty(), "The leaders block cannot be empty");
+        return Iterables.getFirst(leaders(), null);
```

Comment:
`Iterables.getFirst(x)` doesn't exist. I think you're thinking of `getOnlyElement`.

`Iterables.getFirst` calls `iterator().next()` behind the scenes so it should return exactly the same thing. Before, the return value of `iterator().next()` on `Set` was most likely the same on the same JVM but not necessarily the same on different JVMs (major/minor/patch versions, openjdk vs oracle).

Using a List guarantees it to be the same and for it to actually be the first in the List.